bcgov/cas-registration
# fix: update site visited logic for report verification page

### Changes

- Card: https://github.com/bcgov/cas-reporting/issues/199#issuecomment-2524359485
- Updated `ReportVerification` model to include new fields and modify existing ones
- Modified `BaseReportVerificationSchema` to align with new model changes
- Updated `verificationSchema` and `verificationUiSchema` in frontend to reflect new requirements
- Adjusted `VerificationForm` component to handle new form structure

### Acceptance Criteria

- "Site visited" options (Facility X, Other, None) include required fields for threats to independence and verification conclusion
- Facility X and Other options include "Type of site visit" field
- Other option shows additional fields for site name and geographic location
- Backend model and schema align with frontend changes
- Form validation works correctly for all new fields and dependencies

### Test Plan

1. Run migrations and start the application
2. Navigate to the Verification form
3. Test each "Site visited" option:
   - Verify required fields appear for each option
   - Ensure conditional fields show/hide appropriately
4. Submit the form with various combinations of inputs
5. Verify data is correctly saved to the database
6. Check API responses include all new fields

https://www.loom.com/share/8d7c765181ea42daad40b1b9dac8669a?sid=2b18080d-aff5-4fc5-a73c-f392d83265fb
